scripts/ci/ws12_workflow_contracts.py

```
@@ -347,6 +350,38 @@ def step_body(text: str, step_name: str) -> str | None:
+def validate_windows_webui_install_shell(text: str) -> list[str]:
+    """Keep POSIX WebUI setup commands out of PowerShell on Windows."""
+    windows_job = job_body(text, WINDOWS_CLIPPY_JOB)
+    if windows_job is None:
+        return [
+            f"{CODE_STYLE_WORKFLOW}: could not find the {WINDOWS_CLIPPY_JOB!r} job"
+        ]
+    install_step = step_body(windows_job, WEBUI_INSTALL_STEP)
+    if install_step is None:
+        return [
+            f"{CODE_STYLE_WORKFLOW}: {WINDOWS_CLIPPY_JOB!r} has no "
+            f"{WEBUI_INSTALL_STEP!r} step"
```

Ruff (0.16.1)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)